pkg/kubelet/kubelet.go

```diff
@@ -795,6 +802,9 @@ func (kl *Kubelet) initialNodeStatus() (*api.Node, error) {
 			Name:   kl.nodeName,
 			Labels: map[string]string{"kubernetes.io/hostname": kl.hostname},
 		},
+		Spec: api.NodeSpec{
```

Comment:
I am glad to see that we are converging on this. To me that flag --register-schedulable is a temporary hack because we don't have config object defined to generate NodeSpec yet. But to Kubelet, I want NodeSpec is the only source of truth here.